### ISSUE TITLE
[BUG] GIL is not released in some S3 sync wrappers

### ISSUE THREAD

**Issue by leandrodamascena** [ferrumio/pydynox#280]
### Description

Four sync S3 functions call `runtime.block_on()` without releasing the Python GIL. They don't even receive `py: Python<'_>` as a parameter, so they can't call `py.detach()`.

The affected functions are `sync_presigned_url`, `sync_delete_object`, `sync_head_object`, and `sync_save_to_file` in `pydynox/src/s3/operations.rs`.

Other sync S3 functions in the same file (`sync_upload_bytes`, `sync_download_bytes`) correctly use `py.detach()`.

### Steps to Reproduce

```python
import time
from concurrent.futures import ThreadPoolExecutor
from pydynox import DynamoDBClient

client = DynamoDBClient(endpoint_url="http://localhost:8000")

# Upload a test file first
client.sync_s3_upload(bucket="my-bucket", key="test.txt", data=b"hello")

def delete_object(i):
    start = time.time()
    client.sync_s3_delete(bucket="my-bucket", key=f"test-{i}.txt")
    return time.time() - start

# Run 10 deletes in parallel using threads
with ThreadPoolExecutor(max_workers=10) as pool:
    results = list(pool.map(delete_object, range(10)))

# All 10 run one at a time because the GIL is held during block_on()
# Total time ≈ 10 * single_request_time instead of ≈ single_request_time
```

### Expected Behavior

When using `ThreadPoolExecutor`, S3 sync operations should run in parallel. The GIL should be released during network I/O so other threads can proceed.

Total time for 10 parallel requests should be close to 1 request time (network is the bottleneck, not CPU).

### Actual Behavior

All 10 requests run one at a time. Thread 2 waits for thread 1 to finish the full network round-trip before it can start. Total time ≈ 10x a single request.

This happens because `runtime.block_on()` holds the GIL for the entire duration of the async operation. Compare the two patterns in the same file:

```rust
// sync_upload_bytes — CORRECT: releases GIL
pub fn sync_upload_bytes(py: Python<'_>, ...) -> PyResult<...> {
    py.detach(|| {
        runtime.block_on(async { ... })  // GIL released during I/O
    })
}

// sync_presigned_url — BUG: no py parameter, can't release GIL
pub fn sync_presigned_url(
    client: &Client,       // no py: Python<'_> parameter!
    runtime: &Arc<Runtime>,
    bucket: &str,
    key: &str,
    expires_secs: u64,
) -> PyResult<(String, S3Metrics)> {
    runtime.block_on(async { ... })  // GIL held during entire call
}
```

### Environment

- pydynox version: 0.26.0
- Python version: 3.11+
- OS: any
- AWS Region: any

### Additional Context

The fix is to add `py: Python<'_>` parameter and wrap with `py.detach()`:

```rust
pub fn sync_presigned_url(
    py: Python<'_>,  // add this
    client: &Client,
    runtime: &Arc<Runtime>,
    bucket: &str,
    key: &str,
    expires_secs: u64,
) -> PyResult<(String, S3Metrics)> {
    let client = client.clone();
    let bucket = bucket.to_string();
    let key = key.to_string();

    py.detach(|| {
        runtime.block_on(async {
            let result = execute_presigned_url(client, bucket, key, expires_secs)
                .await
                .map_err(S3Exception::new_err)?;
            Ok((result.url, result.metrics))
        })
    })
}
```

Same fix for `sync_delete_object`, `sync_head_object`, and `sync_save_to_file`. Update callers in `s3/client.rs` to pass `py`.

Single-threaded apps are not affected.
